### ISSUE TITLE
Re-elaborate proposal

### ISSUE THREAD

**Issue by JorgeCastilloPrz** [arrow-kt/kotlin#24]
Given our intention to present the KEEP-87 **this week**, we must re-elaborate the proposal in https://github.com/Kotlin/KEEP/pull/87 and also the description in https://github.com/arrow-kt/kotlin/pull/6/. Ideally we'd never refer to FP related examples but more in the line of compile time dependency resolution and validation so it's more thought as a language improvement not just related to FP.

We also need to rename all "implicit" references on KEEP and proposal / texts by "extension".

**Comment by pakoito** [comment]
IT'S HAPPENING

**Comment by soberich** [comment]
@pakoito @JorgeCastilloPrz I have looked all other resources.
> present the KEEP-87 this week...

is there some kind of new video/presentation/article/slides/protocol something which was "happening" 11 days ago or is it just a general "Hooray"?
Interested in KEEP-87, of course.
Thank you in advance.

**Comment by pakoito** [comment]
The branch that'll be sent upstream is getting the last touches.

**Comment by JorgeCastilloPrz** [comment]
Yeps we're quite close to send it, but still not sent. We'll probably make the proper announcements when it's ready.

**Comment by LifeIsStrange** [comment]
@JorgeCastilloPrz any update?
The world needs this feature!

**Comment by JorgeCastilloPrz** [comment]
Hi! The KEEP 87 has not been reviewed by JetBrains during this time. We understand the KEEP process doesn't work as you would expect, most likely because JetBrains priorities and available resources. So we decided to be more productive and pivot to a different direction. That direction will be more powerful and extendable: arrow-meta. We already got it working.

Meta is a library to write compiler plugins available for everyone, and also used by Arrow. It will provide compiler support for things like type class resolution or higher kind emulation, to get those directly supported by the language. Arrow will provide a plugin for this.   

It will also allow us to provide other interesting features like type proofs, which is a more powerful version of Kotlin contracts, and that will allow us to write a library for Refined types.

So the future of Arrow is even brighter now! 🙏
